libs/components/i18n/src/lib/modules/i18n/resources.service.ts

```diff
@@ -143,50 +142,47 @@ export class SkyAppResourcesService {
           return this.#resourcesObsCache[locale];
         }
 
-        resourcesUrl =
+        const resourcesUrl =
           this.#getUrlForLocale(locale) ||
           // Try falling back to the non-region-specific language.
-          this.#getUrlForLocale(locale.substr(0, 2));
-
-        // Finally fall back to the default locale.
-        resourcesUrl =
-          resourcesUrl ||
+          this.#getUrlForLocale(locale.substring(0, 2)) ||
+          // Finally fall back to the default locale.
           this.#getUrlForLocale(this.#localeProvider.defaultLocale);
 
         if (resourcesUrl) {
-          obs =
-            this.#httpObsCache[resourcesUrl] ||
-            this.#http.get<SkyResourceType>(resourcesUrl).pipe(
-              // publishReplay(1).refCount() will ensure future subscribers to
-              // this observable will use a cached result.
-              // https://stackoverflow.com/documentation/rxjs/8247/common-recipes/26490/caching-http-responses#t=201612161544428695958
-              publishReplay(1),
-              refCount(),
-              catchError(() => {
-                // The resource file for the specified locale failed to load;
-                // fall back to the default locale if it differs from the specified
-                // locale.
-                const defaultResourcesUrl = this.#getUrlForLocale(
-                  this.#localeProvider.defaultLocale,
-                );
-
-                if (
-                  defaultResourcesUrl &&
-                  defaultResourcesUrl !== resourcesUrl
-                ) {
-                  return this.#http.get<SkyResourceType>(defaultResourcesUrl);
-                }
-
-                return getDefaultObs();
-              }),
-            );
+          if (!this.#httpObsCache[resourcesUrl]) {
+            this.#httpObsCache[resourcesUrl] = this.#http
+              .get<SkyResourceType>(resourcesUrl)
+              .pipe(
+                share({
+                  connector: () => new ReplaySubject(1),
+                  resetOnError: false,
+                  resetOnComplete: false,
+                  resetOnRefCountZero: false,
+                }),
+                catchError(() => {
+                  // The resource file for the specified locale failed to load;
+                  // fall back to the default locale if it differs from the specified
+                  // locale.
+                  const defaultResourcesUrl = this.#getUrlForLocale(
+                    this.#localeProvider.defaultLocale,
+                  );
+
+                  if (
+                    defaultResourcesUrl &&
+                    defaultResourcesUrl !== resourcesUrl
+                  ) {
+                    return this.#http.get<SkyResourceType>(defaultResourcesUrl);
+                  }
+
+                  return getDefaultObs();
+                }),
+              );
+          }
+          obs = this.#httpObsCache[resourcesUrl];
```

Comment:
☝️ Mind explaining the reasoning for these changes?

Reply:
The `publishReplay` with `refCount` is deprecated, and `share` with `connector` is the recommended replacement.

This was not a required change as it won't break until `rxjs@8`, I was investigating an issue that ended up being caused by the schematic change, but I left these changes since we'd need to address it anyway.